File: core/src/main/java/by/mrrockka/domain/bounty/Bounty.java
package by.mrrockka.domain.bounty;

import by.mrrockka.domain.Person;
import lombok.Builder;
import lombok.NonNull;

import java.math.BigDecimal;

@Builder
@Deprecated(forRemoval = true)
public record Bounty(@NonNull Person from, @NonNull Person to, @NonNull BigDecimal amount) {}

File: core/src/main/java/by/mrrockka/domain/Bounty.kt
package by.mrrockka.domain

import java.math.BigDecimal

data class Bounty(val from: BasicPerson, val to: BasicPerson, val amount: BigDecimal)

fun List<Bounty>.total(): BigDecimal {
    return this.map { it.amount }.fold(BigDecimal.ZERO, BigDecimal::add)
}

fun BountyPlayer.takenToGiven(): Pair<List<Bounty>, List<Bounty>> = bounties.partition { it.to == this.person }